File: server/models/availability.py
from . import shared
from datetime import datetime
db = shared.db

class Availability(db.Model):
    id = db.Column(db.Integer, primary_key=True)
    day = db.Column(db.Integer, nullable=False)
    start_time = db.Column(db.Integer, nullable=False, default=540)
    end_time = db.Column(db.Integer, nullable=False, default=1020)
    created_at = db.Column(db.String(30), nullable=False, default=datetime.utcnow)
    updated_at = db.Column(db.String(30), nullable=False, default=datetime.utcnow)

File: server/models/Availability.js
const { Schema, model } = require("mongoose");

const availabilitySchema = new Schema({
    user:       { type: Schema.Types.ObjectId, ref: "User" },
    day:        { type: Number, required: true },
    startTime:  { type: Number, required: true, default: 540 },
    endTime:    { type: Number, required: true, default: 1020 },
    createdAt:  { type: Date, default: Date.now }, 
    updatedAt:  { type: Date, default: Date.now },  
});

// Format objects returned by Mongoose
availabilitySchema.set("toJSON", {
    transform: (document, returnedObject) => {
        returnedObject.id = returnedObject._id.toString();
        delete returnedObject._id;
        delete returnedObject.__v;
    }
});
  
const Availability = model("Availability", availabilitySchema);

module.exports = Availability;